tidy3d/components/autograd/utils.py

```diff
@@ -2,14 +2,37 @@
 
 import typing
 
-from autograd.tracer import getval
+import autograd.numpy as anp
+import numpy as np
+from autograd.tracer import getval, isbox
+
+
+def _is_traced_objary(x: typing.Any) -> bool:
+    """Check whether x is a traced object array."""
+    return isinstance(x, np.ndarray) and x.size > 0 and isbox(x.flat[0])
+
+
+def is_traced(x: typing.Any) -> bool:
+    """Check whether x is traced (contains an autograd Box)"""
+    return isbox(x) or _is_traced_objary(x)
 
 
 def get_static(x: typing.Any) -> typing.Any:
     """Get the 'static' (untraced) version of some value."""
     return getval(x)
 
 
+def get_box(x: typing.Any) -> typing.Any:
```

Comment:
should we consider just calling this `get_tracer`? to be consistent with our naming?

Reply:
I just used box because that's the autograd terminology. Either is fine. Tracer is probably less confusing though if we stick with one name.

Reply:
yea it makes sense, but maybe we can just use one word for this
* `tracer` = `ArrayBox / Box`
* `traced` = contains a `tracer`
* `get_tracer` = grab the `tracer` from something
* `untraced` = remove the tracer from something
* `insert_tracer` = add a tracer to something?
?